esphome/components/debug/sensor.py

```diff
@@ -47,24 +49,37 @@
         accuracy_decimals=0,
         entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
     ),
+    cv.Optional(CONF_PSRAM): cv.All(
+        cv.only_on_esp32,
+        sensor.sensor_schema(
```

Comment:
This will force fail validation if your config doesnt have `psram` (Either manually added or autoloaded by some components.
```suggestion
    cv.Optional(CONF_PSRAM): cv.All(
        cv.only_on_esp32,
        cv.requires_component("psram"),
        sensor.sensor_schema(
```